core/util/src/main/java/org/eclipse/rdf4j/common/transaction/TransactionSetting.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.common.transaction;
+
+/**
+ * A configuration setting that can be passed at the beginning of a repository transaction to influence behavior within
+ * the scope of that transaction only.
+ *
+ * @author Håvard Ottestad
+ * @author Jeen Broekstra
+ */
+public interface TransactionSetting {
+
+	// FIXME we should perhaps use getURI (with an IRI return type) to uniquely identify settings. Same as what we do
+	// for IsolationLevels.
+	default String getName() {
```

Comment:
Calling it out here as well: For IsolationLevel, we use getURI as the canonical identifier. If we do the same thing generally for TransactionSetting, it would make integration IsolationLevel with the generic handling of TransactionSetttings easier.

Reply:
I think the use of the class/interface canonical name is the best option. Moving the class would not be backwards compatible anyway, and the class canonical name already needs to be unique in the java world. Since this is java code I prefer using a more javaish approach to uniquely identifying a class. It also doesn’t require us to manage namespaces or create IRIs.

Reply:
If even _we_ can't be bothered to manage a namespace.... :) But seriously: fair enough. We should consider making getURI in IsolationLevel deprecated then.